File: F1WM/scripts/add_build_number.py
#! /usr/bin/python

import datetime
import re

date = datetime.datetime.now().strftime("%Y%m%d%H%M")

with open("scripts/version.template.json", "r") as version:
    lines = version.readlines()
with open("version.json", "w") as version:
    for line in lines:
        version.write(re.sub(r'<build_number>', date, line))

File: F1WM/scripts/add_build_number.rb
require 'date'

build_number = DateTime.now.strftime("%Y%m%d%H%M")
branch = 'local'

text = File.read("scripts/version.template.json")
new_contents = text.gsub(/<build_number>/, build_number).gsub(/<branch>/, branch)
File.open("version.json", "w") {|file| file.puts new_contents }